### ISSUE TITLE
Network mount dosent work with kernel (Linux TestPi 5.10.60-v7l+ #1449 SMP Wed Aug 25 15:00:44 BST 2021 armv7l GNU/Linux)

### ISSUE THREAD

**Issue by Blackbox-git** [raspberrypi/linux#4594]
## Issue

When I mount a network share with the command below and the kernel (Linux TestPi 5.10.17-v7l+ #**1414** SMP Fri Apr 30 13:20:47 BST 2021 armv7l GNU/Linux), I can work normally on the drive. (So create folders and files, open and save files).

If I mount this network share with the same command and the kernel (Linux TestPi 5.10.60-v7l+ #**1449** SMP Wed Aug 25 15:00:44 BST 2021 armv7l GNU/Linux), it looks like I can work there as well. However, every time I do this I get the following error:

`sudo mkdir test7`
`mkdir: cannot create directory 'test7': No such file or directory`

or

`sudo touch test.txt`
`touch: cannot touch 'test.txt': No such file or directory`

## Mounting command:

`sudo mount -t cifs -o credentials=/home/pi/.credentials/fritzbox //192.168.111.1/projectbackup/projectbackup/outdoorenclosurePI/ /mnt/backup`

## Output from mount

`//192.168.111.1/projektsicherung/projektsicherung/AußengehegePI/ on /mnt/sicherung type cifs (rw,relatime,vers=3.1.1,cache=strict,username=test user,uid=0,noforceuid,gid=0,noforcegid,addr=192.168.111.1,file_mode=0755,dir_mode=0755,soft,nounix,serverino,mapposix,rsize=65536,wsize=65536,bsize=1048576,echo_interval=60,actimeo=1)`

## Systems

I tested this issue on a Raspberry Pi 2 and Raspberry Pi 4

## Important notice

The only thing I changed during these two points was a kernel update. All other updates were done manually beforehand to avoid any interaction.

**Comment by pelwell** [comment]
This kernel version stamp doesn't make sense:
```
(Linux TestPi 5.10.17-v7l+ #1414 SMP Fri Apr 30 13:20:47 BST 2021 armv7l GNU/Linux)
```
a 5.10.17 build was released in February - the build on Apr 30 was 5.10.33.

**Comment by pelwell** [comment]
You can revert to the Apr 30 5.10.33 with `sudo rpi-update 1340be471e27`, and Feb 25 5.10.17 with `sudo rpi-update f73375d98915`.

**Comment by Blackbox-git** [comment]
If downloaded the Image (Raspberry Pi OS Lite)  from the Homepage (Date May 7th 2021) type the command uname -a an this what i get as output:

`Linux raspberrypi 5.10.17-v7l+ #1414 SMP Fri Apr 30 13:20:47 BST 2021 armv7l GNU/Linux`

With this Version, the mounting works fine. If i do an an full-upgrade the kernel will updated to the version:

`Linux TestPi 5.10.60-v7l+ #1449 SMP Wed Aug 25 15:00:44 BST 2021 armv7l GNU/Linux`

 and the problem begins.

**Comment by pelwell** [comment]
OK - build 1414 is from the `stable` branch, and is a minor change from an earlier build.

Take a look at the page of stable releases: https://github.com/Hexxeh/rpi-firmware/commits/stable
The (usually) 7 character names are values you can give to `rpi-update` to install that version. It would be very helpful if you could try some kernels more recent than that Apr 30th release (`87cbee3`) and report the oldest one that shows the problem.

**Comment by Blackbox-git** [comment]
Ok. Normally every day a backup script is running on my 2 Raspberry Pis. This script normally mount the network drive, copy the data and umount it. 

The script stopped working on 2021-07-30. That is why I have started testing from stable Kernel 03.03.2021.

I have tested all of the following kernel versions. Here are the results.

Can i add a picture to this post?

**Comment by pelwell** [comment]
Just drag and drop it into the comment box.

**Comment by Blackbox-git** [comment]
![Kernel](https://user-images.githubusercontent.com/66961573/134191002-b201fc59-e458-4b8a-bafc-5fbb47f2bdf2.jpg)

**Comment by pelwell** [comment]
So the last known good (in stable) is e7785df (5.10.17, build 1421) and the first bad one (again, in stable) is 4cae6b9 (5.10.50, build 1438). These correspond roughly to https://github.com/Hexxeh/rpi-firmware/commit/f73375d989151f7568e634140d3c4a13044921b2 (good), and https://github.com/Hexxeh/rpi-firmware/commit/4ce61a400c4300df40613d678f7be98f99235f59 (bad) on the [master branch](https://github.com/Hexxeh/rpi-firmware/commits/master). Trying just a few intervening commits to repeatedly halve the distance between good and bad would be a massive help.

**Comment by Blackbox-git** [comment]
Nice great help.

I will check the master branch now in your range. I will post results.

**Comment by Blackbox-git** [comment]
I am done. Here the result of the master branch.

![Kernel2](https://user-images.githubusercontent.com/66961573/134203584-28e32b33-bcfe-451e-a27d-9daa9f45611d.jpg)

**Comment by pelwell** [comment]
That's great - something went wrong between 5.10.17 and 5.10.20. I'll see if there are any obvious candidates.

**Comment by pelwell** [comment]
There is on CIFS commit in that range, but it doesn't look like an obvious candidate. I'm running 5.10.63 and have a writable mount CIFS filesystem mounted.

Reading back through your report there is a mount command that uses English names, but its output is listed in German (including an extended/Unicode character). Which is correct (because I doubt that mount is translating)? Can you experiment with some simpler mount point names/paths?

**Comment by Blackbox-git** [comment]
## Solution for this problem?

Thank you for your answer and your effort.

I have tested a lot and I think I have now found the error myself. Because of your question, I entered the mount command again by hand. Before, I always copied the command from my previous script or the "history" command.

But now to my findings. If I enter the mount command as follows, everything works fine:

`sudo mount -t cifs -o credentials=/home/pi/.credentials/fritzbox //192.168.111.1/projektsicherung/projektsicherung/DisplayPi /mnt/backup`

However, when I enter my previously used or copied command, it just gives the error (mkdir: cannot create directory 'f': No such file or directory):

`sudo mount -t cifs -o credentials=/home/pi/.credentials/fritzbox //192.168.111.1/projektsicherung/projektsicherung/DisplayPi/ /mnt/security`

It's hard to see, but I have a **/** sign after the destination directory. Once this is the case, I can run the mount command but cannot do anything on the drive except display the contents with the ls command.

If I omit the **/** character at the end of the command, everything works as expected.

I am probably the only one who noticed the error, as my very old script had always worked with the / in the target directory and only stopped working after the upgrade of the kernel.

I don't know whether this is a new error in the kernel or whether an earlier error was simply corrected with 5.10.20. I'm not sure.

## New Problem

But now I have another problem. I can copy files to the hard disk. But when I want to display them, for example, via cat, I get the following error:

`cat text.txt`
`cat: text.txt: Stale file handle`

I can reproduce this behaviour on my Linux Mint 20.2 with the specially upgraded kernel 5.11.0-36.

What else do you need from me?

**Comment by pelwell** [comment]
The one change I saw in that timeframe was:
```
b8b65310e4b9 cifs: Set CIFS_MOUNT_USE_PREFIX_PATH flag on setting cifs_sb->prepath.
```
which at least has "path" in the title. Perhaps that was the cause?

With your permission I'll close this issue. Please start a new one for the "stale file handle" problem, otherwise it gets harder to follow.
